### PR TITLE
add mutex for concurrent handler registrations

### PR DESCRIPTION
**Pull request by Maxime2**
We need a mutex to avoid possible data race in concurrent handler registration.

### CODE REVIEW

File: atexit.go [hunk 1]
```diff
@@ -50,6 +52,8 @@ func runHandler(handler func()) {
 }
 
 func runHandlers() {
+	handlersLock.RLock()
```
Comment (tebeka) [inline atexit.go]: I prefer to use [sync.Once](https://golang.org/pkg/sync/#Once) to make sure we don't run handlers twice.

File: atexit.go [hunk 2]
```diff
@@ -30,14 +30,16 @@ import (
...
 )
 
 const (
 	// Version is package version
-	Version = "0.1.0"
+	Version = "0.2.0"
 )
 
 var handlers = []func(){}
+var handlersLock sync.RWMutex
```
Comment (tebeka) [inline atexit.go]: I don't see the value of `RWLock` here. If someone adds a handler while we exit - it won't get executed anyway.

**Comment by tebeka** [comment]
Thanks! Two minor comments. Please  merge/rebase master since I've made some changes (travis support).

File: atexit.go (hunk 1 above)
Comment (Maxime2) [reply atexit.go]: You also need to ensure you do not run handlers at the same time as other goroutine adds another handler.

File: atexit.go (hunk 2 above)
Comment (Maxime2) [reply atexit.go]: RWLock ensures that none of other goroutines adds another handler at the same time and also ensures that handlers do not run while another handler is being added.

File: atexit.go (hunk 1 above)
Comment (tebeka) [reply atexit.go]: Agree

File: atexit.go (hunk 2 above)
Comment (tebeka) [reply atexit.go]: I know. Seems like a overkill since the handlers are going to be read once.

**Pull request merged**